### ISSUE TITLE
Retain `project.build.outputTimestamp` property while trimming BOM

### ISSUE THREAD

**Issue by vy** [apache/logging-parent#121]
`flatten-bom` execution inlines all properties and removes the `<properties>` block. While this is a good BOM practice, it causes following warnings during build:

```
project.build.outputTimestamp property should not be inherited but defined in parent POM from reactor /home/vy/Projects/logging-dist-dev/log4j/3.0.0-beta2-src/.flattened-pom.xml
```

Preferably, `project.build.outputTimestamp` property should be an exempt to BOM trimming and left as is.

**Comment by vy** [comment]
This turned out to be pretty difficult to implement – [fact checked this in #maven](https://the-asf.slack.com/archives/C7Q9JB404/p1712563548050659). AFAIC, we don't have a choice but to keep _only_ the `project.build.outputTimestamp` property to avoid the warning. Though [ElementHandling](https://www.mojohaus.org/flatten-maven-plugin/apidocs/org/codehaus/mojo/flatten/ElementHandling.html) of `flatten-maven-plugin` is pretty limited. I am not able to see a way out except transforming the generated `.flattened-pom.xml` one more time to add the missing `p.b.o` property, and this, AFAIC, necessitates [a very ugly `beanshell-maven-plugin` block](https://github.com/apache/logging-parent/blob/84d70d8f3b8f9219a893a45b454c5eaa478b3481/pom.xml#L584) which I am very reluctant to introduce.

I flirted with the idea of asking help from `flatten-maven-plugin` developers. But judging from their open tickets, the project is barely floating.

In short, due to the complexity involved in solving this problem, I am closing this issue. Feel free to re-open it, if you have a better idea.
